File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = {}
        for n in nums:
            counter[n] = 1 + counter.get(n,0)
        res = [[] for i in range(len(nums) + 1)]
        
        for n,c in counter.items():
            res[c].append(n)

        sol = []
        for i in range(len(res) - 1, 0, -1):
            for n in res[i]:
                sol.append(n)
                if len(sol) == k:
                    return sol



        return result

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] prefix = new int[nums.length];
        int[] suffix = new int[nums.length];
        int[] result = new int[nums.length];
        int j = nums.length - 1;
        for(int i = 0; i<nums.length;i++)
        {
            if(i == 0)
            {
                prefix[0] = nums[0];
                suffix[j] = nums[j];
            }
            else
            {
                prefix[i] = prefix[i-1] * nums[i];
                suffix[j] = suffix[j+1] * nums[j];
            }
            j--;
        }
        for(int i = 0; i < nums.length; i++)
        {
            if(i == 0)
            {
                result[i] = suffix[i+1];
            }
            else if(i == nums.length - 1)
            {
                result[i] = prefix[i -1];
            }
            else
            {
                result[i] = prefix[i-1] * suffix[i+1];
            }
        }
        return result;
    }
}  
